Doc/howto/clinic.rst

```diff
@@ -93,67 +96,148 @@ and it should be able to do many interesting and smart
-
-You should never modify the output portion of an Argument Clinic block.  Instead,
-change the input until it produces the output you want.  (That's the purpose of the
-checksum—to detect if someone changed the output, as these edits would be lost
-the next time Argument Clinic writes out fresh output.)
-
-For the sake of clarity, here's the terminology we'll use with Argument Clinic:
-
-* The first line of the comment (``/*[clinic input]``) is the *start line*.
-* The last line of the initial comment (``[clinic start generated code]*/``) is the *end line*.
-* The last line (``/*[clinic end generated code: checksum=...]*/``) is the *checksum line*.
-* In between the start line and the end line is the *input*.
-* In between the end line and the checksum line is the *output*.
-* All the text collectively, from the start line to the checksum line inclusively,
-  is the *block*.  (A block that hasn't been successfully processed by Argument
-  Clinic yet doesn't have output or a checksum line, but it's still considered
-  a block.)
+will discard the old :term:`output` and write out the new output with a fresh
+:term:`checksum line`.
+If the :term:`input` hasn't changed, the output won't change either.
+
+.. note::
+
+   You should never modify the output of an Argument Clinic block,
+   as any change will be lost in future Argument Clinic runs;
+   Argument Clinic will detect an output checksum mismatch and regenerate the
+   correct output.
+   If you are not happy with the generated output,
+   you should instead change the input until it produces the output you want.
+
+
+.. _clinic-reference:
+
+Reference
+=========
+
+
+.. _clinic-terminology:
+
+Terminology
+-----------
+
+.. glossary::
+
+  start line
+     The line ``/*[clinic input]``.
+     This line marks the beginning of Argument Clinic input.
+     Notice that the *start line* opens a C block comment.
+
+  end line
+     The line ``[clinic start generated code]*/``.
+     This line marks the end of Argument Clinic input.
+     Notice that the *end line* closes the C block comment opened
+     by the *start line*.
+
+  checksum line
+     A line that looks like ``/*[clinic end generated code: ...]*/``.
+     The three dots will be replaced by the actual checksum(s).
+     The checksum line marks the end of Argument Clinic generated code.
+
+  input
+     The text between the :term:`start line` and the :term:`end line`.
+     Notice that the start and end lines open and close a C block comment;
+     the *input* is thus a part of that same C block comment.
+
+  output
+     The text between the :term:`end line` and the :term:`checksum line`.
+
+  block
+     All text from the :term:`start line` to the :term:`checksum line` inclusively.
+
+.. note::
+
+   A block that hasn't been successfully processed by Argument Clinic yet
+   doesn't have *output* or a *checksum line*,
+   but it's still considered a *block*.
+
+
+.. _clinic-cli:
+
+Command-line interface
```

Comment:
`--force`, `--output`, `--verbose` and `--converters` currently have no help output if you run `python Tools/clinic/clinic.py --help`. Could you add the docs you're adding here to the argparsing in `Tools/clinic/clinic.py` as well, so that people will also get this information when they run `python Tools/clinic/clinic.py --help` from the terminal?

(For my own projects, I usually tend not to document the CLI flags (as the ones that are documented can easily go out of date with the ones that are actually available), and just tell people to run `--help` from the command line to see the available flags. But I know that we generally manually document CLI flags in CPython.)

Reply:
That's a good idea.

Reply:
See 01830b5c2f5caaefbd7bd47c45e6a4d336f5d45c